### DISCUSSION TITLE
Snapshot testing

### DISCUSSION THREAD

**Discussion by ivanov-v** [ariakit/ariakit#1143] [Questions] [answered]
Hello. I added a Tooltip to my component, updated the tests and now they point to random IDs. How can I get around this in tests?
```
 -             aria-describedby="id-gn49n2"
 +             aria-describedby="id-rt14l6"

-             id="id-vpjngc"
+             id="id-gp5jvd"
```

**Comment by diegohaz** [comment, marked as answer]
Testing components with Jest Snapshots is generally a terrible idea as you will be testing implementation details like those. So, [my best suggestion is to get rid of those tests](https://twitter.com/diegohaz/status/1506423243158208518?s=21).

If you still want to do that, you can pass your own id to the Tooltip element.

Alternatively, you can also update to React 18.

**Reply by ivanov-v** [reply]
Thanks a lot

**Comment by Anber** [comment]
You can also mock `useId` hook from `ariakit-utils/hooks`:
```
jest.mock("ariakit-utils/hooks", () => {
  const originalModule = jest.requireActual("ariakit-utils/hooks");
  return {
    __esModule: true,
    ...originalModule,
    useId: jest.fn(() => "mocked-id"),
  };
});
```
